core/src/main/scala/cats/data/HashMap.scala

```diff
@@ -0,0 +1,1196 @@
+ * @param hashKey the [[cats.kernel.Hash]] instance used for hashing keys.
+ */
+final class HashMap[K, +V] private[data] (private[data] val rootNode: HashMap.Node[K, V])(implicit val hashKey: Hash[K])
+    extends HashMapCompat[K, V] {
+
+  /**
+    * An iterator for this map that can be used only once.
+    *
+    * @return an iterator that iterates through the key-value pairs of this map.
+    */
+  final def iterator: Iterator[(K, V)] =
+    new HashMap.Iterator(rootNode)
+
+  /**
+    * A reverse iterator for this map that can be used only once.
+    *
+    * @return an iterator that iterates through this map in the reverse order of [[HashMap#iterator]].
+    */
+  final def reverseIterator: Iterator[(K, V)] =
+    new HashMap.ReverseIterator(rootNode)
+
+  /**
+    * The size of this map.
+    *
+    * @return the number of elements in this map.
+    */
+  final def size: Int = rootNode.size
+
+  /**
+    * Tests whether the map is empty.
+    *
+    * @return `true` if the map contains no elements, `false` otherwise.
+    */
+  final def isEmpty: Boolean = size == 0
+
+  /**
+    * Tests whether the map is not empty.
+    *
+    * @return `true` if the map contains at least one element, `false` otherwise.
+    */
+  final def nonEmpty: Boolean = !isEmpty
+
+  /**
+    * Apply `f` to each key-value pair for its side effects.
+    *
+    * @param f the function to apply to each key-value pair.
+    */
+  final def foreach[U](f: (K, V) => U): Unit =
+    rootNode.foreach(f)
+
+  /**
+    * Test whether the map contains `key`.
+    *
+    * @param key the key to check for map membership.
+    * @return `true` if the map contains `key`, `false` otherwise.
+    */
+  final def contains(key: K): Boolean =
+    rootNode.contains(key, improve(hashKey.hash(key)), 0)
+
+  /**
+    * Get the value associated with `key` in this map.
+    *
+    * @param key the key to check for map membership.
+    * @return A [[scala.Some]] containing the value if present, else [[scala.None]].
+    */
+  final def get(key: K): Option[V] =
+    rootNode.get(key, improve(hashKey.hash(key)), 0)
+
+  /**
+    * Get the value associated with `key` in this map, or `default` if not present.
+    *
+    * @param key the key to check for map membership.
+    * @param default the value to use in case `key` is not present.
+    * @return the value if present, else `default`.
+    */
+  final def getOrElse[VV >: V](key: K, default: => VV): VV =
+    get(key).getOrElse(default)
+
+  /**
+    * Creates a new map with an additional key-value pair, unless the key is already present,
+    * in which case the value for `key` is replaced by `value`.
+    *
+    * @param key the key to be added.
+    * @param value the value to be added.
+    * @return a new map that contains all key-value pairs of this map and that also contains a mapping from `key` to `value`.
+    */
+  final def add[VV >: V](key: K, value: VV): HashMap[K, VV] = {
```

Comment:
doesn't scala use `updated` here? Can we use that here?

Reply:
I took the lead here from `Chain` which uses the verb `append` instead of the stdlib symbolic `+` or past tense `appended`. Definitely happy to change it 👍

Reply:
I renamed this to `updated` and also `remove` to `removed` for consistency with the 2.13.x standard library maps